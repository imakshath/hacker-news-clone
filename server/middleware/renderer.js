import React from 'react'
import ReactDOMServer from 'react-dom/server'
import { Provider as ReduxProvider } from 'react-redux';
// import our main App component
import App from '../../src/App';

const path = require("path");
const fs = require("fs");

export default (store) => (req, res, next) => {
    // point to the html file created by CRA's build tool
    const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');

    fs.readFile(filePath, 'utf8', (err, htmlData) => {
        if (err) {
            return res.status(404).end()
        }

        const html = ReactDOMServer.renderToString(
            <ReduxProvider store={store}>
                <App />
            </ReduxProvider>
        );

        // inject the rendered app into our html and send it
        return res.send(
            htmlData.replace(
                '<div id="root"></div>',
                `<div id="root">${html}</div>`
            )
        );
    });
}