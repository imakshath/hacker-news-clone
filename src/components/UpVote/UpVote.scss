.upvote-arrow {
    width: 10px;
    height: 10px;
    border: 0px;
    margin: 3px 2px 6px;
    background: url(https://news.ycombinator.com/grayarrow.gif) no-repeat;
    cursor: pointer;
}