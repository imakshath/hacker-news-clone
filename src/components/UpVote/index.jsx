import React from 'react';
import './UpVote.scss';

function UpVote({ onVoteUpStory, storyId }) {
    const handleVoteUp = React.useCallback(() => {
        onVoteUpStory(storyId);
    }, [onVoteUpStory, storyId]);

    return (
        <div data-testid={`upvote-${storyId}`} className="upvote-arrow" onClick={handleVoteUp}></div>
    );
}

export default React.memo(UpVote);
