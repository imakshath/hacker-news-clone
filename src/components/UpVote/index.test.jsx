import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import UpVote from '.';

afterEach(cleanup);

describe('<UpVote /> component', () => {
    let wrapper;
    const props = {
        storyId: "123",
        onVoteUpStory: jest.fn()
    };
    
    beforeEach(() => {
        wrapper = render(
            <UpVote {...props} />
        );
    });

    test('should render without crash', () => {
        expect(wrapper.container).toBeDefined();
    });

    test('should get by testid', () => {
        expect(wrapper.getByTestId('upvote-123')).toBeDefined();
    });

    test('should match snapshot', () => {
        const tree = renderer.create(<UpVote />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})