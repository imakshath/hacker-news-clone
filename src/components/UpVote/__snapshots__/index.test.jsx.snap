// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpVote /> component should match snapshot 1`] = `
<div
  className="upvote-arrow"
  data-testid="upvote-undefined"
  onClick={[Function]}
/>
`;
